### DISCUSSION TITLE
Are Jetstream push subscribers completely deprecated?

### DISCUSSION THREAD

**Discussion by rafsawicki** [nats-io/nats.deno#652] [General] [answered]
Coming from NATS Streaming, which only had push consumers, we originally migrated our consumers in Jetstream as push consumers as well. Now that we're doing some code cleanup, it seems that the original Jetstream `subscribe` method is deprecated, alongside the `deliver_subject` parameter in the new API for creating consumers. I cannot find any non-deprecated way of creating push consumers. 

Does it mean, that push subscribers are going to be completely removed from NATS at some point, in favor of pull subscribers, and they should be completely avoided? Or maybe I simply missed the proper API that should be used to create them?

**Comment by aricart** [comment, marked as answer]
The push subscriber will remain there, however you should not use it - the new consumers abstraction - https://github.com/nats-io/nats.deno/blob/main/jetstream.md#consuming-messages does exactly what you want, but it does it behind a pull consumer. In terms of your code, it will feel like a push, but will have all the benefits of a pull consumer. Use this instead please.
